src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -63,106 +123,53 @@ def from_config_dict(self, config_dict):
 
         ext_job_list.convertToCReference(None)
 
         # Create varlist
-        site_config_env_vars: EnvironmentVarlist = site_config._get_env_var_list()
-        environment_vars = config_dict.get(ConfigKeys.SETENV, [])
-
-        for elem in environment_vars:
-            site_config_env_vars.setenv(elem[ConfigKeys.NAME], elem[ConfigKeys.VALUE])
-
-        return site_config
-
-    def __repr__(self):
-        if not self._address():
-            return "<SiteConfig()>"
-        return (
-            "SiteConfig(config_dict={"
-            + f"{ConfigKeys.INSTALL_JOB}: ["
-            + ", ".join(str(job) for job in self.get_installed_jobs())
-            + "],"
-            + f"{ConfigKeys.SETENV}: {self._get_env_var_list()}"
-            + "})"
-        )
-
-    @property
-    def config_file(self):
-        return self._get_config_file()
+        env_vars = EnvironmentVarlist()
+        dict_vars = config_dict.get(ConfigKeys.SETENV, [])
 
-    def get_installed_jobs(self):
-        return self._get_installed_jobs().setParent(self)
+        for elem in dict_vars:
+            env_vars.setenv(elem[ConfigKeys.NAME], elem[ConfigKeys.VALUE])
 
-    def get_license_root_path(self) -> str:
-        return self._get_license_root_path()
+        dict_paths = config_dict.get("UPDATE_PATH", [])
 
-    @classmethod
-    def getLocation(cls) -> str:
-        return cls._get_location()
+        for elem in dict_paths:
+            env_vars.update_path(elem[ConfigKeys.NAME], elem[ConfigKeys.VALUE])
 
-    def free(self):
-        self._free()
-
-    def __eq__(self, other):
-        self_job_list = self.get_installed_jobs()
-        other_job_list = other.get_installed_jobs()
-
-        if self._get_env_var_list() != other._get_env_var_list():
-            return False
-
-        if set(other_job_list.getAvailableJobNames()) != set(
-            self_job_list.getAvailableJobNames()
-        ):
-            return False
+        if site_config_content is not None:
+            SiteConfig._add_config_content(
+                license_root_path, site_config_content, ext_job_list, env_vars
+            )
 
-        if len(other_job_list.getAvailableJobNames()) != len(
-            self_job_list.getAvailableJobNames()
-        ):
-            return False
-
-        for job_name in other_job_list.getAvailableJobNames():
-
-            if (
-                other_job_list[job_name].get_config_file()
-                != self_job_list[job_name].get_config_file()
-            ):
-                return False
-
-            if (
-                other_job_list[job_name].get_stderr_file()
-                != self_job_list[job_name].get_stderr_file()
-            ):
-                return False
-
-            if (
-                other_job_list[job_name].get_stdout_file()
-                != self_job_list[job_name].get_stdout_file()
-            ):
-                return False
-        return True
+        return SiteConfig(ext_job_list, env_vars)
```

Comment:
i'm a bit of a noob when it comes to classes in python, so this is an honest question - would you use `cls` instead of `SiteConfig` here? same question in line 171 above

Reply:
also in the three relevant lines in `from_config_content`

Reply:
I guess you could use cls. Is that preferable?